src/core/src/lib.rs

```diff
@@ -238,3 +239,107 @@ pub trait Device: Sized {
     /// Cleanup unused resources. This should be called between frames. 
     fn cleanup(&mut self);
 }
+
+
+/// An `Instance` holds per-application state for a specific backend
+pub trait Instance {
+    /// Associated `Adapter` type.
+    type Adapter: Adapter;
```

Comment:
Why should we have `Instance` and `Adapter` be separate types? They both kinda converge to representing a backend.